aten/src/ATen/native/Normalization.cpp

```diff
@@ -23,6 +25,195 @@ namespace {
   }
 }
 
+// TensorAccessor when it is defined to work around undefined...
+template <typename scalar_t>
+static TensorAccessor<scalar_t, 1> conditional_accessor_1d(const Tensor& t) {
+  if (! t.defined()) {
+    return TensorAccessor<scalar_t, 1>(nullptr, nullptr, nullptr);
+  }
+  return t.accessor<scalar_t, 1>();
+}
+
+
+template<typename scalar_t>
+std::tuple<Tensor,Tensor,Tensor> batch_norm_cpu_template(const Tensor& input, const Tensor& weight, const Tensor& bias,
+			       const Tensor& running_mean, const Tensor& running_var, bool train, double momentum, double eps) {
+
+  using accscalar_t = at::acc_type<scalar_t, false>;
+  Tensor output = at::native::empty_like(input);
```

Comment:
Today I learned that you should always do `at::empty_like`. Yes it saves dispatch code, but it makes changing the dispatching implementation harder.

Reply:
OK, I can adapt to that. :)